providers/summary/scancode.js

```diff
@@ -90,12 +85,49 @@ class ScanCodeSummarizer {
     }
   }
 
-  _addLicenseFiles(file, declaredLicenses) {
-    // Look for license files at the root of the scanned code
-    // TODO enhance in the future to cover more license management strategies.
-    if (!['license', 'license.txt', 'license.md', 'license.html'].includes(file.path.toLowerCase())) return
-    if (!file.licenses) return
-    file.licenses.forEach(license => declaredLicenses.add(license.spdx_license_key))
+  _summarizeDeclaredLicenseInfo(files) {
+    const declaredLicenses = new Set()
+    for (const fileIdx in files) {
+      const file = files[fileIdx]
+      const pathArray = file.path.split('/')
+      const baseName = last(pathArray)
+      const isLicense = ['license', 'license.txt', 'license.md', 'license.html'].includes(baseName.toLowerCase())
+      if (isLicense && file.licenses) {
+        // Find first license file and return
+        file.licenses.forEach(license => declaredLicenses.add(license.spdx_license_key))
+        return declaredLicenses
+      }
+    }
+    // Did not find license file so returning empty set (no declared license)
+    return declaredLicenses
+  }
+
+  _summarizePackageInfo(files) {
+    const packageLicenses = new Set()
+    for (const fileIdx in files) {
```

Comment:
suggest `forEach` for looping over arrays (here and elsewhere) when you do not need the index.